components/accelerator/nvidia/gpu-counts/component.go

```diff
@@ -0,0 +1,280 @@
+		if err != nil {
+			log.Logger.Warnw("error getting count of lspci", "error", err)
+		} else {
+			log.Logger.Infow("count of lspci", "count", cr.CountLspci)
+		}
+	}
+
+	if cr.CountNVML != thresholds.Count {
+		cr.health = apiv1.HealthStateTypeUnhealthy
+		cr.reason = fmt.Sprintf("nvidia gpu count mismatch (nvml %d, expected %d)", cr.CountNVML, thresholds.Count)
+		log.Logger.Errorw(cr.reason, "count_lspci", cr.CountLspci, "count_nvml", cr.CountNVML, "expected", thresholds.Count)
+		return cr
+	}
+
+	cr.health = apiv1.HealthStateTypeHealthy
+	cr.reason = fmt.Sprintf("nvidia gpu count matching thresholds (%d)", thresholds.Count)
+	return cr
+}
+
+const reasonThresholdNotSetSkipped = "GPU count thresholds not set, skipping"
+
+var _ components.CheckResult = &checkResult{}
+
+type checkResult struct {
+	ProductName string `json:"product_name"`
+	CountLspci  int    `json:"count_lspci"`
+	CountNVML   int    `json:"count_nvml"`
+
+	// timestamp of the last check
+	ts time.Time
+	// error from the last check with "ibstat" command and other operations
+	err error
+
+	// tracks the healthy evaluation result of the last check
+	health apiv1.HealthStateType
+	// tracks the suggested actions for the last check
+	suggestedActions *apiv1.SuggestedActions
+	// tracks the reason of the last check
+	reason string
+}
+
+func (cr *checkResult) ComponentName() string {
+	return Name
+}
+
+func (cr *checkResult) String() string {
+	if cr == nil {
+		return ""
+	}
+	if cr.ProductName == "" {
+		return "no data"
+	}
+
+	buf := bytes.NewBuffer(nil)
+	table := tablewriter.NewWriter(buf)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.Append([]string{"Product Name", cr.ProductName})
+	table.Append([]string{"Count lspci", fmt.Sprintf("%d", cr.CountLspci)})
+	table.Append([]string{"Count NVML", fmt.Sprintf("%d", cr.CountNVML)})
+	table.Render()
+
+	return buf.String()
+}
+
+func (cr *checkResult) Summary() string {
+	if cr == nil {
+		return ""
+	}
+	return cr.reason
+}
+
+func (cr *checkResult) HealthStateType() apiv1.HealthStateType {
+	if cr == nil {
+		return ""
+	}
+	return cr.health
+}
+
+func (cr *checkResult) getSuggestedActions() *apiv1.SuggestedActions {
+	if cr == nil {
+		return nil
+	}
+	return cr.suggestedActions
+}
+
+func (cr *checkResult) getError() string {
+	if cr == nil {
+		return ""
+	}
+	if cr.err != nil {
+		return cr.err.Error()
+	}
+	return ""
+}
+
+func (cr *checkResult) HealthStates() apiv1.HealthStates {
+	if cr == nil {
+		return apiv1.HealthStates{
+			{
+				Time:      metav1.NewTime(time.Now().UTC()),
+				Component: Name,
+				Name:      Name,
+				Health:    apiv1.HealthStateTypeHealthy,
+				Reason:    "no data yet",
+			},
+		}
+	}
+
+	state := apiv1.HealthState{
+		Time:             metav1.NewTime(cr.ts),
+		Component:        Name,
+		Name:             Name,
+		Health:           cr.health,
+		Reason:           cr.reason,
+		Error:            cr.getError(),
+		SuggestedActions: cr.getSuggestedActions(),
```

Comment:
What's the suggested action? In some cases, the GPU may be lost and can be recovered by rebooting. In some case, it cannot be recovered even after rebooting.. It is better to suggest restarting N times first, and then suggest hardware inspection